build/management/commands/build_rfaa_models.py

```diff
@@ -0,0 +1,1013 @@
+                    if created:
+                        self.logger.info('Created structure type {}'.format(st))
+                except IntegrityError:
+                    st = StructureType.objects.get(slug=structure_type_slug)
+                struct.structure_type = st
+            else:
+                self.logger.warning('No structure type specified in PDB file {}'.format(sd['pdb']))
+
+#################################################
+
+            # insert into plain text fields
+            if 'preferred_chain' in sd:
+                struct.preferred_chain = sd['preferred_chain']
+            else:
+                self.logger.warning('Preferred chain not specified for structure {}'.format(sd['pdb']))
+            if 'resolution' in sd:
+                struct.resolution = float(sd['resolution'])
+            else:
+                struct.resolution = None
+                self.logger.warning('Resolution not specified for structure {}. Setting as null'.format(sd['pdb']))
+            if 'publication_date' in sd:
+                struct.publication_date = sd['publication_date']
+            else:
+                sd['publication_date'] = build_date = date.today()
+                struct.publication_date = sd['publication_date']
+                self.logger.warning('Publication date not specified for structure {}. Defaulting at today'.format(sd['pdb']))
+
+            struct.annotated = True
+            struct.refined = False
+            struct.stats_text = None
+
+            #Resolution and
+            # save structure before adding M2M relations
+            struct.save()
+
+            print('BEFORE CREATING LIGAND')
+
+            # Create LigandPeptideStructure object
+            try:
+                # Get the existing Ligand object
+                ligand = Ligand.objects.get(inchikey=sd['ligand_id'])
+                
+                # Try to get existing LigandPeptideStructure or create a new one
+                ligand_peptide_structure, created = LigandPeptideStructure.objects.get_or_create(
+                    structure=struct,
+                    ligand=ligand,
+                    chain='B',
+                    defaults={'model': None}  # Set model to None
+                )
+                
+                if created:
+                    print(f"Created new LigandPeptideStructure for structure {struct.pdb_code.index} and ligand {ligand.name}")
+                else:
+                    print(f"Found existing LigandPeptideStructure for structure {struct.pdb_code.index} and ligand {ligand.name}")
+            
+            except Ligand.DoesNotExist:
+                print(f"Error: Ligand with id {sd['ligand_id']} does not exist in the database.")
+            except Exception as e:
+                print(f"Error creating LigandPeptideStructure: {str(e)}")
+
+####################################################
+            Rotamer.objects.filter(structure=struct, pdbdata=pdbdata).delete()
```

Comment:
I'm not sure what this is here. I don't think you have built the rotamer objects but even if you did why delete them?